### PR TITLE
tree-wide: drop udev_device struct and use sd_device instead

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/libudev/libudev-monitor.c [hunk 1]
```diff
@@ -702,6 +703,19 @@ _public_ struct udev_device *udev_monitor_receive_device(struct udev_monitor *ud
...
+int udev_monitor_receive_sd_device(struct udev_monitor *udev_monitor, sd_device **ret) {
+        _cleanup_(udev_device_unrefp) struct udev_device *udev_device = NULL;
+
+        assert(ret);
+
+        udev_device = udev_monitor_receive_device(udev_monitor);
+        if (!udev_device)
+                return -errno;
+
+        *ret = sd_device_ref(udev_device->device);
```
Comment (keszybz) [inline src/libudev/libudev-monitor.c:715]: At some point we probably want to do the reverse, and make `udev_monitor_receive_device` a wrapper around `udev_monitor_receive_sd_device`.

File: src/backlight/backlight.c [hunk 2]
```diff
@@ -417,13 +446,13 @@ int main(int argc, char *argv[]) {
...
+                if (validate_device(device) == 0) {
                         unlink(saved);
                         return EXIT_SUCCESS;
                 }
 
-                value = udev_device_get_sysattr_value(device, "brightness");
-                if (!value) {
+                r = sd_device_get_sysattr_value(device, "brightness", &value);
+                if (r < 0 || isempty(value)) {
                         log_error("Failed to read system 'brightness' attribute");
```
Comment (keszybz) [inline src/backlight/backlight.c]: log_error_errno…

File: src/mount/mount-tool.c [hunk 3]
```diff
@@ -1335,16 +1325,11 @@ static int discover_device(void) {
...
-        udev = udev_new();
-        if (!udev)
-                return log_oom();
-
-        d = udev_device_new_from_devnum(udev, 'b', st.st_rdev);
-        if (!d)
-                return log_oom();
+        r = sd_device_new_from_devnum(&d, 'b', st.st_rdev);
+        if (r < 0)
+                return log_error_errno(r, "Failed to get device from device number");
```
Comment (keszybz) [inline src/mount/mount-tool.c]: "…: %m"

File: src/network/networkd-link.c [hunk 4]
```diff
@@ -3201,13 +3201,18 @@ int link_add(Manager *m, sd_netlink_message *message, Link **ret) {
...
+                r = sd_device_new_from_device_id(&device, ifindex_str);
+                if (r < 0) {
+                        log_link_warning_errno(link, r, "Could not find device: %m");
                         goto failed;
                 }
 
-                if (udev_device_get_is_initialized(device) <= 0) {
+                r = sd_device_get_is_initialized(device, &initialized);
+                if (r < 0) {
+                        log_link_warning_errno(link, r, "Could not determine the device is initialized or not: %m");
```
Comment (keszybz) [inline src/network/networkd-link.c]: Could not determine **whether** the device …

File: src/libudev/libudev-monitor.c (hunk 1 above)
Comment (yuwata) [reply src/libudev/libudev-monitor.c:715]: Yeah. It was done in the previous PR, but this PR does not include cleanups for libudev... I will do that later in another PR.

File: src/backlight/backlight.c (hunk 2 above)
Comment (yuwata) [reply src/backlight/backlight.c]: ~~`r` may be non-negative. no?~~

**Comment by yuwata** [comment]
@keszybz Thank you for the review. Two log messages you mentioned are fixed.

File: src/backlight/backlight.c (hunk 2 above)
Comment (yuwata) [reply src/backlight/backlight.c]: Also fixed.

**Pull request merged**